pandas/core/nanops.py

```diff
@@ -865,6 +867,17 @@ def reduction(
                 result.fill(np.nan)
             except (AttributeError, TypeError, ValueError):
                 result = np.nan
+        elif (
+            is_object_dtype(dtype)
```

Comment:
why do you need all of these condtions? this is complicated

Reply:
Only looking at objects is for the reason above, `values.ndim == 1` is because I think we can get here even if `values` is not vector-valued, which is more or less what we're assuming when we just mask them out (if we don't include this we can get test failures when we have an all-object `DataFrame`), `mask.any()` is because this function already works if no values are missing so there's no reason to do masking, and `mask is not None` is to please `mypy` (we've already guaranteed that `mask` isn't `None` for objects above but `mypy` doesn't know this).

I can try to be a bit more explicit in the comments if that would be helpful.